packages/worker/src/worker.ts

```diff
@@ -993,260 +998,261 @@ export class Worker {
+      map(() => ({
+        activation: coresdk.workflow_activation.WorkflowActivation.create({
+          runId: activations$.key,
+          jobs: [{ removeFromCache: Worker.SELF_INDUCED_SHUTDOWN_EVICTION }],
+        }),
+        synthetic: true,
+      })),
+      takeUntil(activations$.pipe(last(undefined, null)))
+    );
+    const activations$$ = activations$.pipe(map((activation) => ({ activation, synthetic: false })));
+    return merge(activations$$, evictions$).pipe(
+      tap(() => {
+        this.numInFlightActivationsSubject.next(this.numInFlightActivationsSubject.value + 1);
+      }),
+      mergeMapWithState(this.handleActivation.bind(this), undefined),
+      tap(({ close }) => {
+        this.numInFlightActivationsSubject.next(this.numInFlightActivationsSubject.value - 1);
+        if (close) {
+          activations$.close();
+          this.numCachedWorkflowsSubject.next(this.numCachedWorkflowsSubject.value - 1);
+        }
+      }),
+      takeWhile(({ close }) => !close, true /* inclusive */)
+    );
+  }
+
+  /**
+   * Process a single activation to a completion.
+   */
+  protected async handleActivation(
+    state: WorkflowWithLogAttributes | undefined,
+    { activation, synthetic }: { activation: coresdk.workflow_activation.WorkflowActivation; synthetic: boolean }
+  ): Promise<{
+    state: WorkflowWithLogAttributes | undefined;
+    output: { completion?: Uint8Array; close: boolean };
+  }> {
+    try {
+      const removeFromCacheIx = activation.jobs.findIndex(({ removeFromCache }) => removeFromCache);
+      const close = removeFromCacheIx !== -1;
+      const jobs = activation.jobs;
+      if (close) {
+        const asEvictJob = jobs.splice(removeFromCacheIx, 1)[0].removeFromCache;
+        if (asEvictJob) {
+          this.evictionsEmitter.emit('eviction', {
+            runId: activation.runId,
+            evictJob: asEvictJob,
+          } as EvictionWithRunID);
+        }
+      }
+      activation.jobs = jobs;
+      if (jobs.length === 0) {
+        this.log.trace('Disposing workflow', {
+          ...(state ? state.logAttributes : { runId: activation.runId }),
+        });
+        await state?.workflow.dispose();
+        if (!close) {
+          throw new IllegalStateError('Got a Workflow activation with no jobs');
+        }
+        const completion = synthetic
+          ? undefined
+          : coresdk.workflow_completion.WorkflowActivationCompletion.encodeDelimited({
+              runId: activation.runId,
+              successful: {},
+            }).finish();
+        return { state: undefined, output: { close, completion } };
+      }
+
+      if (state === undefined) {
+        // Find a workflow start job in the activation jobs list
+        const maybeStartWorkflow = activation.jobs.find((j) => j.startWorkflow);
+        if (maybeStartWorkflow !== undefined) {
+          state = await this.getWorkflowState(activation, maybeStartWorkflow.startWorkflow || undefined);
+        } else {
```

Comment:
Please rename that function. It is not about getting a workflow's state, it is about instantiating the workflow execution context (choose the word you want: context/sandbox/isolate). Also, at this point, `maybeStartWorkflow.startWorkflow` can't be undefined (it was checked the line before).

Reply:
Thanks. I've changed the function name to `createWorkflowWithLogAttributes()` and I've renamed the local variable `state` to `workflow`.

Reply:
Just `createWorkflow()`. The `WithLogAttributes` is a pretty insignificant by-product.

Reply:
Done!